From 6638f8125e420f4ef1e95fcece1e2b31c3781708 Mon Sep 17 00:00:00 2001
From: Gustavo Duarte <gduarte@redhat.com>
Date: Tue, 16 Dec 2014 15:09:58 -0500
Subject: [powerpc] perf/hv-24x7: Use kmem_cache_free

Message-id: <1418742601-6870-2-git-send-email-gduarte@redhat.com>
Patchwork-id: 102100
O-Subject: [RHEL7.1 PATCH BZ 1171795 1/4] powerpc/perf/hv-24x7: Use kmem_cache_free
Bugzilla: 1171795
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

BZ 1171795
Upstream Status: d6589722846a57a4ddf7af595a7f854ff5180950

commit d6589722846a57a4ddf7af595a7f854ff5180950
Author: Himangi Saraogi <himangi774@gmail.com>
Date:   Tue Jul 22 23:40:19 2014 +0530

    powerpc/perf/hv-24x7: Use kmem_cache_free

    Free memory allocated using kmem_cache_zalloc using kmem_cache_free
    rather than kfree.

    The Coccinelle semantic patch that makes this change is as follows:

    // <smpl>
    @@
    expression x,E,c;
    @@

     x = \(kmem_cache_alloc\|kmem_cache_zalloc\|kmem_cache_alloc_node\)(c,...)
     ... when != x = E
         when != &x
    ?-kfree(x)
    +kmem_cache_free(c,x)
    // </smpl>

    Signed-off-by: Himangi Saraogi <himangi774@gmail.com>
    Acked-by: Julia Lawall <julia.lawall@lip6.fr>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/perf/hv-24x7.c b/arch/powerpc/perf/hv-24x7.c
index 9f284db..0f948f2 100644
--- a/arch/powerpc/perf/hv-24x7.c
+++ b/arch/powerpc/perf/hv-24x7.c
@@ -150,7 +150,7 @@ e_free:
   pr_err("h_get_24x7_catalog_page(ver=%lld, page=%lld) failed:"
          " rc=%ld\n",
          catalog_version_num, page_offset, hret);
- kfree(page);
+ kmem_cache_free(hv_page_cache, page);
 
  pr_devel("catalog_read: offset=%lld(%lld) count=%zu "
    "catalog_len=%zu(%zu) => %zd\n", offset, page_offset,
-- 
1.7.1